### ISSUE TITLE
python demo 为什么加载shufflenet,而不是自己下载的nanodet_m.pth文件，跑出来效果明显不好。

### ISSUE THREAD

**Issue by zoufangyu1987** [RangiLyu/nanodet#58]
如题

**Comment by zoufangyu1987** [comment]
![Uploading IMG_20201203_232511.jpg…]()

**Comment by leozicai** [comment]
前面是初始化模型，后面 load_model_weight(model, ckpt, logger) 就将nanodet_m.pth作为权重加载到初始化的模型上了。。。。

**Comment by zoufangyu1987** [comment]
@leozicai 
程序前几行的输出为:
model size is 1.0x
init weights...
=> loading pretrained model https://download.pytorch.org/models/shufflenetv2_x1-5666bf0f80.pth

这个看起来像是加载了shufflenetv2_x1-5666bf0f80.pth模型，并不是nanodet_m.pth模型，面具检测结果不太像正常的结果，效果不好。

![2020-12-04 09-57-52屏幕截图](https://user-images.githubusercontent.com/30693954/101111977-526db000-3617-11eb-9457-17fd82e5405c.jpg)

**Comment by CBIR-LL** [comment]
使用的shufflenet做backbone，对其初始化的时候加载了你提到的shufflenet的模型。最终load还是nanno的模型。你看一眼nanodet/model/backbone/shufflenetv2.py就明白了。这种问题，只要你看一眼代码，就不会来问了。

**Comment by zoufangyu1987** [comment]
@CBIR-LL 
谢谢，我主要是看到效果不怎么好，所以有这个疑问，我还没有看代码调试，THANKS！

**Comment by CBIR-LL** [comment]
@zoufangyu1987 效果嘛，和tiny打平手吧，所以不能要求太多，毕竟是要快。我在自己的数据上还行，iou0.5话，map0.949.

**Comment by zoufangyu1987** [comment]
@CBIR-LL 
上图中的效果正常吗？是不是它的类型不支持还是怎么的，感觉效果确实不好，当然我还没有基于自己的训练数据集finetune.非常感谢您！

**Comment by CBIR-LL** [comment]
@zoufangyu1987 你需要在自己的数据集上训练和测试。你用其他的数据来测试coco训练出的模型是不太公平的。

**Comment by zoufangyu1987** [comment]
@CBIR-LL 
知道了，感谢！

**Comment by connorchen** [comment]
> @zoufangyu1987 你需要在自己的数据集上训练和测试。你用其他的数据来测试coco训练出的模型是不太公平的。

这个我就不太理解了，就比如车，难道和coco数据集中的有什么不一样？

**Comment by leozicai** [comment]
> 
> 
> > @zoufangyu1987 你需要在自己的数据集上训练和测试。你用其他的数据来测试coco训练出的模型是不太公平的。
> 
> 这个我就不太理解了，就比如车，难道和coco数据集中的有什么不一样？

这个基于coco训练出来的模型，coco80类，用这么小个模型拟合80类。而我用全是车的数据集，只有car一个类，训练出一个专门检测car的模型，你觉得那个效果好？打个比方，让你只学英语，和让你学语文数学英语物理化学生物历史地理政治，你对英语的掌握程度相同吗？

**Comment by CBIR-LL** [comment]
@connorchen 如果你是第一天做目标检测，就能理解你的问题。如果不是，建议你自己思考思考。训练集和测试集同分布，检测效果是要好的，而测试集是不同抽样分布的，你觉得会好吗？举个简单的例子：训练集合都是正脸，测试集合是有俯仰角的各种角度人脸，你觉得检测效果会好？

**Comment by connorchen** [comment]
> > > @zoufangyu1987 你需要在自己的数据集上训练和测试。你用其他的数据来测试coco训练出的模型是不太公平的。
> > 
> > 
> > 这个我就不太理解了，就比如车，难道和coco数据集中的有什么不一样？
> 
> 这个基于coco训练出来的模型，coco80类，用这么小个模型拟合80类。而我用全是车的数据集，只有car一个类，训练出一个专门检测car的模型，你觉得那个效果好？打个比方，让你只学英语，和让你学语文数学英语物理化学生物历史地理政治，你对英语的掌握程度相同吗？



> @connorchen 如果你是第一天做目标检测，就能理解你的问题。如果不是，建议你自己思考思考。训练集和测试集同分布，检测效果是要好的，而测试集是不同抽样分布的，你觉得会好吗？举个简单的例子：训练集合都是正脸，测试集合是有俯仰角的各种角度人脸，你觉得检测效果会好？

膜拜下

**Comment by zoufangyu1987** [comment]
@connorchen 
我用作者提供的模型在自己训练数据集上finetune了一个epoch，效果还是可以的，模型没有问题。

**Comment by eeric** [comment]
解释的挺清楚。
